evm/src/cpu/control_flow.rs

```diff
@@ -68,14 +68,18 @@ pub fn eval_packed_generic<P: PackedField>(
         lv.is_cpu_cycle * is_native_instruction * (lv.is_kernel_mode - nv.is_kernel_mode),
     );
 
-    // If a non-CPU cycle row is followed by a CPU cycle row, then the `program_counter` of the CPU
-    // cycle row is route_txn (the entry point of our kernel) and it is in kernel mode.
+    // If a non-CPU cycle row is followed by a CPU cycle row, then:
+    //  - the `program_counter` of the CPU cycle row is `route_txn` (the entry point of our kernel),
+    //  - execution is in kernel mode, and
+    //  - the stack is empty.
     let pc_diff =
         nv.program_counter - P::Scalar::from_canonical_usize(KERNEL.global_labels["route_txn"]);
     yield_constr.constraint_transition((lv.is_cpu_cycle - P::ONES) * nv.is_cpu_cycle * pc_diff);
     yield_constr.constraint_transition(
         (lv.is_cpu_cycle - P::ONES) * nv.is_cpu_cycle * (nv.is_kernel_mode - P::ONES),
     );
+    yield_constr
```

Comment:
nit: let's factor out `(lv.is_cpu_cycle - P::ONES) * nv.is_cpu_cycle` now that it's used 3x